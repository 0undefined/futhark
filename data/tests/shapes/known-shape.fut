// An existing variable can be used as a shape declaration.

fun [[int,!k],!n] main(int n, int m, int k) =
  let a = replicate(n, iota(m)) in
  zipWith(fn [int,!k] (int i, [int,!m] r) =>
            let x = reduce(+, 0, r)
            in map(+i, map(+x, iota(k))),
          iota(n), a)
