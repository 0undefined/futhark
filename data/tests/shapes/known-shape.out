[[6, 7, 8, 9, 10, 11, 12, 13],
 [7, 8, 9, 10, 11, 12, 13, 14],
 [8, 9, 10, 11, 12, 13, 14, 15],
 [9, 10, 11, 12, 13, 14, 15, 16],
 [10, 11, 12, 13, 14, 15, 16, 17]]
